src/functions/contactsMenu/index.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+type ContactsMenuCallback = () => void
+
+interface ContactsMenuHookProps {
+	id: string;
+	callback: ContactsMenuCallback;
+}
+
+declare global {
+	interface Window {
+		_vue_contacts_menu_hooks: Record<string, ContactsMenuHookProps>;
+	}
+}
+
+if (!window._vue_contacts_menu_hooks) {
+	window._vue_contacts_menu_hooks = {}
+}
+
+const registerContactsMenuHook = (id: string, callback: ContactsMenuCallback) => {
```

Comment:
I think for our syntax it is more common to directly export.
Also using `function` will hoist the declaration of it which prevents issues in general.

```suggestion
export function registerContactsMenuHook(id: string, callback: ContactsMenuCallback): void {
```

Reply:
Yeah, I also like exporting directly and not having const closures more.

Done.